### PR TITLE
feat(kucoin) - watchBidsAsks

### LINKED ISSUE #22038: Kucoin Pro: support for top 1 best ask and bid websocket
### Preliminary Checks

- [X] I have already searched for [existing issues](https://github.com/ccxt/ccxt/issues) and confirmed that this issue is not a duplicate

### Is your feature request related to a problem? Please describe

_No response_

### Describe the solution you'd like

Kucoin supports subscribing to the best ask and bid of up to 100 symbols per topic, so basically 1 level deep order book. See:
https://www.kucoin.com/docs/websocket/spot-trading/public-channels/level1-bbo-market-data

Ccxt only supports subscribing to best 5, best 50, or the complete order book. The best 5 and 50 websockets are pushed at a 100ms interval. The top 1 best ask and bid are pushed at a 10ms interval, and in some use cases the top 1 is more than enough and thus preferred due to low latency. The format of the output looks the same as the top 5 and 50 so it should be fairly simple to implement based on the others.

### Describe alternatives you've considered

_No response_

### PR DESCRIPTION
**Pull request by ttodua**
fix #22038

```
npm run cli.js -- kucoin watchBidsAsks "['LINK/USDT','ADA/USDT']"

> ccxt@4.2.89 cli.js
> node ./examples/js/cli.js kucoin watchBidsAsks ['LINK/USDT','ADA/USDT']

2024-04-04T12:12:39.803Z
Node.js: v20.11.1
CCXT v4.2.89
kucoin.watchBidsAsks (LINK/USDT,ADA/USDT)
{
  'LINK/USDT': {
    symbol: 'LINK/USDT',
    timestamp: 1712232762737,
    datetime: '2024-04-04T12:12:42.737Z',
    ask: 17.8935,
    askVolume: 53.9,
    bid: 17.8926,
    bidVolume: 26.95
  }
}
{
  'ADA/USDT': {
    symbol: 'ADA/USDT',
    timestamp: 1712232763282,
    datetime: '2024-04-04T12:12:43.282Z',
    ask: 0.5823,
    askVolume: 1342.36,
    bid: 0.5821,
    bidVolume: 4360
  }
}
```

### CODE REVIEW

**Comment by carlosmiei** [comment]
@ttodua can you check the build pls/ 
image

**Comment by ttodua** [comment]
sure, will make this PR ready soon and tag you

**Comment by ttodua** [comment]
recoded few things. in kucoinfutures, there was more realistic fetchTicker endpoint (while the currently used one was only orderbook's best bid-ask channel, which is exactly what we call watchBidsAsks, and not anyway 'ticker' - which should contain h/l/c/o/v datas). so, the v1-ticker endpoint still contains more ticker data than 'best-bid-ask' channel. 
users should be aware that they should use `watchBidsAsks` for that function and not `watchTickers`.

**Comment by ttodua** [comment]
@carlosmiei idk on travis, but tests pass on local, you can runtest locally to ensure

**Comment by ttodua** [comment]
watchOrderBook is somewhat not working as expected, but not related to this PR in any way, will push a fix PR separately

File: ts/src/pro/kucoinfutures.ts [hunk 1]
```diff
@@ -675,6 +780,9 @@ export default class kucoinfutures extends kucoinfuturesRest {
         //     }
         //
         const id = this.safeString (message, 'id');
+        if (!(id in client.subscriptions)) {
```
Comment (carlosmiei) [inline ts/src/pro/kucoinfutures.ts:783]: @ttodua why are you doing this? client.subscriptions is indexed by the messagehash not the subscription id

Comment (carlosmiei) [reply ts/src/pro/kucoinfutures.ts:783]: if you want to skip it using the id you need to do the indexBy first (done below)

Comment (ttodua) [reply ts/src/pro/kucoinfutures.ts:783]: ah, accidentally copied from kucoin.ts, they have similar messages, but different handling, yes.

**Comment by carlosmiei** [comment]
@ttodua build is still broken 
image

**Comment by ttodua** [comment]
sorry, I had a windwhirl here probably.
eventually, it is ready

**Comment by carlosmiei** [comment]
DEMO

```
Python v3.11.7
CCXT v4.2.90
kucoin.watchBidsAsks(['BTC/USDT', 'LTC/USDT'])
{'BTC/USDT': {'ask': 67636.2,
              'askVolume': 0.58494175,
              'average': None,
              'baseVolume': None,
              'bid': 67636.1,
              'bidVolume': 1.01532964,
              'change': None,
              'close': None,
              'datetime': '2024-04-06T13:26:12.670Z',
              'high': None,
              'info': {'data': {'asks': ['67636.2', '0.58494175'],
                                'bids': ['67636.1', '1.01532964'],
                                'timestamp': 1712409972670},
                       'subject': 'level1',
                       'topic': '/spotMarket/level1:BTC-USDT',
                       'type': 'message'},
              'last': None,
              'low': None,
              'open': None,
              'percentage': None,
              'previousClose': None,
              'quoteVolume': None,
              'symbol': 'BTC/USDT',
              'timestamp': 1712409972670,
              'vwap': None}}
```
```
Python v3.11.7
CCXT v4.2.90
kucoinfutures.watchBidsAsks(['BTC/USDT:USDT', 'LTC/USDT:USDT'])
{'BTC/USDT:USDT': {'ask': 67629.5,
                   'askVolume': 1923.0,
                   'average': None,
                   'baseVolume': None,
                   'bid': 67629.4,
                   'bidVolume': 935.0,
                   'change': None,
                   'close': None,
                   'datetime': '2024-04-06T13:26:49.542Z',
                   'high': None,
                   'info': {'data': {'bestAskPrice': '67629.5',
                                     'bestAskSize': 1923,
                                     'bestBidPrice': '67629.4',
                                     'bestBidSize': 935,
                                     'sequence': 1700782795012,
                                     'symbol': 'XBTUSDTM',
                                     'ts': 1712410009542000000},
                            'sn': 1700782795012,
                            'subject': 'tickerV2',
                            'topic': '/contractMarket/tickerV2:XBTUSDTM',
                            'type': 'message'},
                   'last': None,
                   'low': None,
                   'open': None,
                   'percentage': None,
                   'previousClose': None,
                   'quoteVolume': None,
                   'symbol': 'BTC/USDT:USDT',
                   'timestamp': 1712410009542,
                   'vwap': None}}
```

**Pull request merged**